cpp/modules/deck.gl/layers/src/solid-polygon-layer/solid-polygon-layer.cpp

```diff
@@ -0,0 +1,198 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "./solid-polygon-layer.h"  // NOLINT(build/include)
+
+#include "deck.gl/core.h"
+
+using namespace deckgl;
+using namespace lumagl;
+
+const std::vector<const Property*> propTypeDefs = {
+    new PropertyT<bool>{
+        "filled", [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->filled; },
+        [](JSONObject* props, bool value) { return dynamic_cast<SolidPolygonLayer::Props*>(props)->filled = value; },
+        true},
+    new PropertyT<bool>{
+        "extruded",
+        [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->extruded; },
+        [](JSONObject* props, bool value) { return dynamic_cast<SolidPolygonLayer::Props*>(props)->extruded = value; },
+        false},
+    new PropertyT<bool>{
+        "wireframe",
+        [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->wireframe; },
+        [](JSONObject* props, bool value) { return dynamic_cast<SolidPolygonLayer::Props*>(props)->wireframe = value; },
+        false},
+    new PropertyT<bool>{
+        "material",
+        [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->material; },
+        [](JSONObject* props, bool value) { return dynamic_cast<SolidPolygonLayer::Props*>(props)->material = value; },
+        true},
+    new PropertyT<bool>{
+        "normalize",
+        [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->normalize; },
+        [](JSONObject* props, bool value) { return dynamic_cast<SolidPolygonLayer::Props*>(props)->normalize = value; },
+        true},
+    new PropertyT<float>{
+        "elevationScale",
+        [](const JSONObject* props) { return dynamic_cast<const SolidPolygonLayer::Props*>(props)->elevationScale; },
+        [](JSONObject* props, bool value) {
+          return dynamic_cast<SolidPolygonLayer::Props*>(props)->elevationScale = value;
+        },
+        1.0}};
+
+auto SolidPolygonLayer::Props::getProperties() const -> const Properties* {
+  static Properties properties{Properties::from<SolidPolygonLayer::Props>("SolidPolygonLayer", propTypeDefs)};
+  return &properties;
+}
+
+void SolidPolygonLayer::initializeState() {
+  auto polygon = std::make_shared<arrow::Field>("instancePolygons", arrow::fixed_size_list(arrow::float32(), 4));
+  auto getPolygon = std::bind(&SolidPolygonLayer::getPolygonData, this, std::placeholders::_1);
```

Comment:
Just curious, is `std::bind` being used here because of issues with accessing `this` from within a lambda function?

Reply:
I think @ilijapuaca  ran into problems and did it like this, but yes if possible now, let's use lambdas. You can capture `this` directly or just any outside scope references in a lambda by value `[this](...) {}` or `[=]() {}`.

Reply:
I believe there was another reason for this that I can't recall right now. @randypalusz you can give lambdas a go per suggestion from @ibgreen. We have a retain cycle that we need to deal with regardless, still not really sure how to go about that in C++